query_server/spi/src/catalog/mod.rs

```diff
@@ -32,6 +33,10 @@ pub trait MetaData: Send + Sync {
     fn create_table(&self, name: &str, table: Arc<dyn TableProvider>) -> Result<()>;
     fn create_database(&self, name: &str, database: DatabaseSchema) -> Result<()>;
     fn database_names(&self) -> Vec<String>;
+    fn describe_database(&self, name: &str) -> Result<Output>;
+    fn describe_table(&self, table_name: &str) -> Result<Output>;
+    fn show_database(&self) -> Result<Output>;
+    fn show_table(&self, database_name: &str) -> Result<Output>;
```

Comment:
I think the object schema that the metadata interface should return, not the result set.

```
eg.
    fn describe_database(&self, name: &str) -> Result<DatabaseSchema>;

```

Reply:
Somewhere should return Result<Output>.

Reply:
This is metadata interface